powershell_kernel/powershell_proxy.py

```diff
@@ -40,8 +40,10 @@ def __init__(self, repl):
         self.expected_output_len = 0
 
         # this is a hack to detect when we stop processing this input
-        self.run_command('function prompt() {"^"}')
-
+        # Run command is enumerable as a result we need to pass in prompt properly
+        for temp in self.run_command('function prompt() {"^"}'):
+            continue
+        
     def run_command(self, input):
```

Comment:
Let's add comment, something like:
# returns a generator that yields string messages as the are returned from powershell via stdout